src/gamekit.cs

```diff
@@ -853,6 +853,22 @@ interface GKLocalPlayer
 		Action<NSViewController, NSError> AuthenticateHandler { get; set; }
 #endif
 
+		[NoWatch, NoTV, Mac (12,0), iOS (15,0), MacCatalyst (15,0)]
+		[Export ("isPresentingFriendRequestViewController")]
+		bool IsPresentingFriendRequestViewController { get; }
+
+#if !MONOMAC
+		[NoWatch, NoTV, NoMac, iOS (15,0), NoMacCatalyst]
+		[Export ("presentFriendRequestCreatorFromViewController:error:")]
+		bool PresentFriendRequestCreator (UIViewController viewController, [NullAllowed] out NSError error);
+#endif
+
+#if MONOMAC
+		[NoWatch, NoTV, NoiOS, Mac (12,0), NoMacCatalyst]
+		[Export ("presentFriendRequestCreatorFromWindow:error:")]
+		bool PresentFriendRequestCreatorFromWindow ([NullAllowed] NSWindow window, [NullAllowed] out NSError error);
+#endif
```

Comment:
Is the #if needed? Also, also, you have a #if !MONOMAC and a #if MONOMAC what to use a:
```suggestion
#if MONOMAC
		[NoWatch, NoTV, NoiOS, Mac (12,0), NoMacCatalyst]
		[Export ("presentFriendRequestCreatorFromWindow:error:")]
		bool PresentFriendRequestCreatorFromWindow ([NullAllowed] NSWindow window, [NullAllowed] out NSError error);
#else
		[NoWatch, NoTV, NoMac, iOS (15,0), NoMacCatalyst]
		[Export ("presentFriendRequestCreatorFromViewController:error:")]
		bool PresentFriendRequestCreator (UIViewController viewController, [NullAllowed] out NSError error);
#endif
```

Reply:
Is it not needed? I think I got a build error when I did not use the #ifs.
Also these two methods are not related so I thought the separate #ifs would be better in case these were ever moved or anything?

Reply:
1. What error did you get?
2. No, we want to reduce the number of #if else it is very hard to review and confusing. Preprocessor directives make things harder.

Reply:
Okay, I'll build again and see which error it was!

Reply:
The error:
```
gamekit.cs(866,61): error CS0246: The type or namespace name 'NSWindow' could 
not be found (are you missing a using directive or an assembly reference?)
```

Reply:
Hm, that errors is coming when you compile for iOS, which makes sense since NSWindow is part of AppKit and not present on iOS. So, the two methods are related, and I would do the following:

1. use the #if/#else/#endif
2. Rename PresentFriendRequestCreatorFromWindow to PresentFriendRequestCreator since FromWindow is redundant. 

Both methods are probably the same one but working around the fact that UIKit and AppKit are not the same.

Reply:
Ah okay that makes sense thanks! They do look related and I just missed it!